packages/medusa/src/repositories/price-list.ts

```diff
@@ -1,5 +1,111 @@
-import { EntityRepository, Repository } from "typeorm"
+import { flatten, groupBy, map } from "lodash"
+import {
+  Brackets,
+  EntityRepository,
+  FindManyOptions,
+  FindOperator,
+  OrderByCondition,
+  Repository,
+} from "typeorm"
 import { PriceList } from "../models/price-list"
+import { PriceListStatus, PriceListType } from "../types/price-list"
 
+type CustomOptions = {
```

Comment:
suggestion: Extract to util since it seems to be quite generic

Reply:
```
type CustomFindOptions<TModel extends BaseEntity> = {
  select?: FindManyOptions["select"]
  where?: FindManyOptions["where"] & {
    status?: FindOperator<TModel extends BaseEntity>[]
    type?: FindOperator<TModel extends BaseEntity>[]
  }
  order?: OrderByCondition
  skip?: number
  take?: number
} 
```

Reply:
I don't think it should go to util but into the type directory in the `common` file.

Reply:
Pushed a fix, but still need your opinion on it @adrien2p, it looks like this: 
```typescript
export interface CustomFindOptions<TModel, InKeys extends keyof TModel> {
  select?: FindManyOptions<TModel>["select"]
  where?: FindManyOptions<TModel>["where"] &
    {
      [P in InKeys]?: TModel[P][]
    }
  order?: OrderByCondition
  skip?: number
  take?: number
}
```

We can't really have 100% of what @olivermrbl suggested because the `status` and `type` fields are specific to the `PriceList` entity, but it seems like what we usually do is override some properties' type(s) by an array to later implement a logical "OR" or "IN" (i.e: get the price lists with type "override" OR "sale"). That's why I added another generic to the type so that consumers of the type specify which property(ies) be augmented as array

Reply:
Nice job @zakariaelas it looks good to me if I have properly understood the needs 😀